src/marshmallow/fields.py

```diff
@@ -187,7 +188,19 @@ def __init__(
             raise ValueError("'missing' must not be set for required fields.")
         self.required = required
         self.missing = missing
-        self.metadata = metadata
+
+        # intentionally shallow-copy so that more data can be added without changing
+        # the original mapping object, but objects can be referenced from field
+        # metadata
+        self.metadata = dict(metadata) if metadata is not None else {}
+        if additional_metadata:
+            self.metadata.update(additional_metadata)
```

Comment:
Maybe

```suggestion
        metadata = metadata or {}
        self.metadata = {**metadata, **additional_metadata}
        if additional_metadata:
```

Reply:
I like this better! Taking it.

Also, I didn't know that `**` could be used on an arbitrary Mapping until today. So thanks for teaching me something new. 😄